embedded-graphics/src/mock_display.rs

```diff
@@ -1,24 +1,125 @@
-use crate::drawable::{Dimensions, Pixel};
+use crate::drawable::Pixel;
 use crate::pixelcolor::BinaryColor;
 use crate::prelude::*;
-use crate::{Drawing, SizedDrawing};
+use crate::Drawing;
+use core::{
+    cmp::PartialEq,
+    fmt::{self, Write},
+    iter,
+};
+
+const SIZE: usize = 64;
 
 /// Mock display for use in tests and some doc examples. Do not use directly!
-#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
-pub struct MockDisplay<C>(pub [[C; 24]; 16]);
+#[derive(Clone)]
+pub struct MockDisplay<C>([Option<C>; SIZE * SIZE])
+where
+    C: PixelColor;
+
+impl<C> MockDisplay<C>
+where
+    C: PixelColor,
+{
+    /// Creates a new empty mock display.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Returns a iterator over all pixels on the display.
+    pub fn pixels<'a>(&'a self) -> impl Iterator<Item = Option<C>> + 'a {
+        self.0.into_iter().copied()
+    }
+}
+
+impl<C> MockDisplay<C>
+where
+    C: PixelColor + ColorMapping<C>,
+{
+    /// Creates a new mock display from a character pattern.
+    ///
+    /// The color pattern is specified by a slice of string slices. Each string slice represents
+    /// a row of pixels and every character a single pixel.
+    ///
+    /// A space character in the pattern represents a pixel which wasn't modified by any
+    /// drawing routine and is left in the default state. All other characters are converted
+    /// by implementations of the `ColorMapping` trait.
```

Comment:
Please turn `ColorMapping` into a doc link

```suggestion
    /// by implementations of the [`ColorMapping`](./trait.ColorMapping.html) trait.
```

To make the char conversion easier for other contributors to use, I'd like to see a quick bullet point list in either this doc comment or another describing which characters map to what colour. I'm fine with just supporting `BinaryColor` for tests so that mapping should be easy to document. Maybe in the `//!` module docs?